### DISCUSSION TITLE
okd latest version fails in vsphere UPI with network customizations.

### DISCUSSION THREAD

**Discussion by GaneshKumarKasiviswanathan** [okd-project/okd#816] [Q&A]
I am following Craig's steps to install OKD in vsphere with cutom network and bootstrap is failing.
DNS is through named - resolvs fine
DHCP is through pfsense
LB - haproxy
web server - httpd

Please let me what information is needed to troubleshoot

**Comment by rvanderp3** [comment]
Would you be able to attach a [log bundle](https://docs.okd.io/latest/installing/installing-troubleshooting.html)?

**Reply by GaneshKumarKasiviswanathan** [reply]
[log-bundle-20210817233154.tar.gz](https://github.com/openshift/okd/files/7002260/log-bundle-20210817233154.tar.gz)

**Reply by GaneshKumarKasiviswanathan** [reply]
[bootstrapinstallerror.txt](https://github.com/openshift/okd/files/7002270/bootstrapinstallerror.txt)
[bootstrapvmerror.txt](https://github.com/openshift/okd/files/7002272/bootstrapvmerror.txt)

**Reply by GaneshKumarKasiviswanathan** [reply]
Please check and help

**Reply by rvanderp3** [reply]
The log-bundle didn't collect any logs from the control-plane nodes.  This can indicate that control plane nodes failed to ignite.  Can you check the console on the control plane nodes and see if there are any errors.  I can see the machine config server is up on the bootstrap:

~~~
I0817 17:33:49.677003       1 bootstrap.go:37] Version: machine-config-daemon-4.6.0-202006240615.p0-736-gfc6f6c6f (fc6f6c6fcc912000db20fc255918021c3bd225a8)
I0817 17:33:49.680038       1 api.go:72] Launching server on :22623
I0817 17:33:49.680611       1 api.go:72] Launching server on :22624
~~~

If the control plane nodes are able to reach port 22623 on the bootstrap node via the load balancer, they should have completed ignition.

**Reply by GaneshKumarKasiviswanathan** [reply]
This is what happening in all master nodes:

![image](https://user-images.githubusercontent.com/65657984/129843206-53e6ab8a-acad-4160-84c1-ea51dd9e81b8.jpg)

**Reply by rvanderp3** [reply]
It appears that the master nodes are not able to perform a DNS lookup.  Note that port 53 is refusing the connection.  That could mean that the DNS server the masters are configured for is incorrect or may not be reachable.  Can you confirm that the master nodes have an IP?  Can you ping them?

**Reply by GaneshKumarKasiviswanathan** [reply]
Thanks for helping me. 
I am using pfsense for DHCP. Given static ip's for all master, worker, bootstrap, okd services nodes in the pfsense. DNS is configured using named and able to ping, resolve ip including reverse PTR records for all nodes. From the bootstrap node and okd4-services node I am able to ping the master nodes IP. When I check in vCenter no ips are assigned to master during bootstrap. Only the bootstrap node gets the ip assigned from DHCP. Anything specific that you want me to check and share screenshot?

**Reply by rvanderp3** [reply]
No problem at all, happy to help.  Can you enable a serial port on one of the master nodes and direct the output to a file?
![image](https://user-images.githubusercontent.com/49568690/129909128-f829b9a5-e7c8-4b25-b079-ebda00feece3.png)

**Reply by GaneshKumarKasiviswanathan** [reply]
Ok, I’ll do that and update you. In the meanwhile sharing the dns and other configurations for your reference

**Reply by GaneshKumarKasiviswanathan** [reply]
![image](https://user-images.githubusercontent.com/65657984/129910504-97611280-0a0e-45fe-a924-c54bb0dff928.jpg)

**Reply by GaneshKumarKasiviswanathan** [reply]
![image](https://user-images.githubusercontent.com/65657984/129910637-6a769cc9-2b36-4cdf-9b67-637ae7cd6a1d.jpg)

**Reply by GaneshKumarKasiviswanathan** [reply]
![image](https://user-images.githubusercontent.com/65657984/129910970-cc91ac88-7fd8-42f4-8b09-57cc7f31db5c.jpg)

**Reply by GaneshKumarKasiviswanathan** [reply]
cat db.okd.local

![image](https://user-images.githubusercontent.com/65657984/129911199-8710cd95-2c11-4f0e-805f-1a501e9f95cc.jpg)

**Reply by GaneshKumarKasiviswanathan** [reply]
![image](https://user-images.githubusercontent.com/65657984/129913735-c9692570-24d2-4ae1-87d8-b0142145bc51.jpg)

**Reply by GaneshKumarKasiviswanathan** [reply]
Hi @rvanderp3 
I do not see that option of serial port while creating master VM

![image](https://user-images.githubusercontent.com/65657984/129916516-cfb4efe0-c400-496f-a68e-4ff2e504d927.jpg)

**Reply by rvanderp3** [reply]
I am not sure why that would be missing.  It is present on my vCenter.  Since we can't get the serial logs, a screen shot of the master a earlier in the process after the errors start appearing would be helpful.  The only concrete error we have so far is that the master cant reach the DNS server.

**Reply by GaneshKumarKasiviswanathan** [reply]
How about my other DNS, DHCP, install-config? Does they look good? any other way to check?

**Reply by rvanderp3** [reply]
As best I can tell they are fine. I would try booting a linux VM with the same networking configuration as one of the masters and see if you can curl the URL mentioned in the error.  There appears to be some network issue.

**Reply by GaneshKumarKasiviswanathan** [reply]
Same issue again, not sure where the issue? Spent lot of time on this. Is there any other east way to setup okd on vm home lab?

**Reply by GaneshKumarKasiviswanathan** [reply]
Is so share steps or link please

**Reply by rvanderp3** [reply]
The issue appears to be related to the network unfortunately based on the limited information we have.  What was the result of trying the curl with the linux VM?  If the result was the same, troubleshooting the network is the best path forward in my opinion.
